### ISSUE TITLE
Question: tvOS: When using tvos remote, UISearchController keyboard events are not propagated to UISearchBar's Text Input Field

### ISSUE THREAD

**Issue by keshavkaul** [react-native-community/react-native-tvos#47]
I am trying to Add Search capability in my tvOS App and am facing an issue where when using the tvos remote, the keyboard events are not propagated to the SearchBar's Text Input Field.

Typing on the laptop works though.

I have created a reproducible test repo for your review. (Branch-`poc/search-controller`)
https://github.com/keshavkaul/react-native-tvos-poc/tree/poc/search-controller

In the repo, I have created a Native UI Component called `RNSearchControllerView`
The native code is present under `src/search-controller/ios/Classes`

Any help is most welcome!

My guess for the issue is that react is intercepting the remote's events and not correctly propagating to the target UI

**Comment by dlowder-salesforce** [comment]
React does intercept remote events, in `RCTTVRemoteHandler.m`.  You might need to modify that file in order to allow your search controller to get the events.

**Comment by dlowder-salesforce** [comment]
@keshavkaul were you able to resolve this issue?

**Comment by keshavkaul** [comment]
@dlowder-salesforce No, I didn't find time to resolve this, but I implemented a quick fix to natively present a search controller as a modal instead of embedding the controller in react native view

**Comment by hscheidl** [comment]
I'm facing the same issue, but I can't figure out how to avoid `RCTTVRemoteHandler.m` from intercepting remote events just for the search controller. Has anybody integrated successfully the search bar? Is there a work-around besides showing it as a modal?

**Comment by zsoltbokor** [comment]
Hi, we also have `UISearchController` used in our project and had issues mostly with remote with touch.
Sometimes touch event (gestures) were not propagated to the keyboard.
After dubugging it found out that there is a `RCTTouchHandler` registered in `React/Base/RCTRootContentView.m` so basically 2 gesture handlers were registered. 
`RCTTouchHandler` has internally a function `canBePreventedByGestureRecognizer ` which was returning true in our case.

The `RCTTVRemoteHandler .m` already has gesture handler implementation so in case of `RCTRootContentView` we just skip to register the `RCTTouchHandler`.

**Comment by douglowder** [comment]
@zsoltbokor if you are using recent RNTV (0.72 latest, or 0.73), the `RCTTouchHandler` has been removed for Apple TV, see https://github.com/react-native-tvos/react-native-tvos/pull/601 .

**Comment by ashnfb** [comment]
I was able to track this down to the UITapGestureHandler for RCTTVRemoteHandler.js, specifically the UIPressTypeSelect which is cancelling touches from propogating

By calling TVEventControl.disableGestureHandlersCancelTouches(),  the search keyboard responds to remote press, and  characters appear in the search text field!

cc @zsoltbokor

**Comment by ashnfb** [comment]
@CodingByJerez , here's what I have implemented for Apple TV Search:  it requires a lot of workarounds in the native code to work

hope it is of help

[Archive.zip](https://github.com/user-attachments/files/21627970/Archive.zip)

**Comment by ruskomilchev** [comment]
About [logicwind/react-native-tvos-keyboard](https://github.com/logicwind/react-native-tvos-keyboard?utm_source=chatgpt.com) — I had to go through a similar patching experience: vibe-coding Swift to patch feature gaps as they came up. The main issues were focus handling, the keyboard not reflecting theme changes (dark vs. light mode), and placement of search results in grid mode. I ended up handling the grid positioning with position: absolute in React Native.

This is definitely one of the Expo UI Swift components on my wish list. I also checked that most major native tvOS apps use the native keyboard (not custom keyboard UI).
